daffodil-core/src/main/scala/org/apache/daffodil/dsom/ElementBase.scala

```diff
@@ -653,13 +653,16 @@ trait ElementBase
     }
   }
 
+  private lazy val optionBinaryNubmerRep = findPropertyOption("binaryNumberRep")
+
   private def getImplicitAlignmentInBits(thePrimType: PrimType, theRepresentation: Representation): Int = {
     (theRepresentation, thePrimType) match {
       case (Representation.Text, PrimType.HexBinary) => Assert.impossible("type xs:hexBinary with representation='text'")
       case (Representation.Text, _) => knownEncodingAlignmentInBits
       case (Representation.Binary, PrimType.Float | PrimType.Boolean) => 32
       case (Representation.Binary, PrimType.Double) => 64
-      case (Representation.Binary, _) => binaryNumberRep match {
+      case (Representation.Binary, PrimType.HexBinary) => 8
+      case (Representation.Binary, _) if (optionBinaryNumberRep.isDefined) => binaryNumberRep match {
```

Comment:
Is the optionBinaryNumberRep.isDefine check needed here? If the prim type is hexBInary it'll hit the new case you added and will never try to get optionBInaryNumberRep, I think? There's also a case PrimType.HexBinary => 8 at the bottom of this match case that I think is impossible to hit with the above change? Hard to tell with this complex nesting of match cases...

Reply:
I think this rats nest of match cases was made by me, so I took a stab at greatly simplifying the logic here and I think it is much cleaner now.  I ran the test suite with these changes and everything passes, but give it a look to make sure I didn't miss anything obvious.